### PR TITLE
Internal: Conversion Center - Contact Button - UI control tweaks

### PR DESCRIPTION
**Pull request by willhowat**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Contact Button - UI updates to controls

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 1]
```diff
@@ -1429,12 +1432,13 @@ protected function add_style_top_bar_section(): void {
 					[
 						'label' => esc_html__( 'Close Button', 'elementor' ),
 						'type' => Controls_Manager::HEADING,
-						'separator' => false,
+						'separator' => $close_and_background_partial_divider,
```
Comment (mserino) [inline modules/conversion-center/base/widget-contact-button-base.php:1435]: @willhowat can we add this to the config instead? So you would add it at the top of this file in the `style` -> `top_bar_section` as a variable like `has_close_and_background_separator` and set it to `false` by default
and you would access it like this `$config['style']['top_bar_section']['has_close_and_background_separator']` from the variant that needs it

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 2]
```diff
@@ -1448,34 +1452,42 @@ protected function add_style_top_bar_section(): void {
 					'condition' => $this->get_platform_color_condition( [
 						'style_top_bar_colors' => 'custom',
 					] ),
+					'separator' => $close_and_background_partial_divider,
```
Comment (mserino) [inline modules/conversion-center/base/widget-contact-button-base.php:1455]: see comment above

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 3]
```diff
@@ -1448,34 +1452,42 @@ protected function add_style_top_bar_section(): void {
...
+					[
+						'label' => esc_html__( 'Background', 'elementor' ),
+						'type' => Controls_Manager::HEADING,
+						'separator' => $close_and_background_partial_divider,
+						'condition' => $this->get_platform_color_condition( [
+							'style_top_bar_colors' => 'custom',
+						] ),
+					]
+				);
+				$close_and_background_partial_divider = false;
```
Comment (mserino) [inline modules/conversion-center/base/widget-contact-button-base.php:1476]: see comment above

File: modules/conversion-center/base/widget-contact-button-base.php [hunk 4]
```diff
@@ -1448,34 +1452,42 @@ protected function add_style_top_bar_section(): void {
...
-					'label' => esc_html__( 'Background', 'elementor' ),
+					'label' => esc_html__( 'Background Color', 'elementor' ),
 					'type' => Controls_Manager::COLOR,
 					'selectors' => [
 						'{{WRAPPER}} .e-contact-buttons' => '--e-contact-buttons-top-bar-bg: {{VALUE}}',
 					],
 					'condition' => $this->get_platform_color_condition( [
 						'style_top_bar_colors' => 'custom',
 					] ),
+					'separator' => $close_and_background_partial_divider,
```
Comment (mserino) [inline modules/conversion-center/base/widget-contact-button-base.php:1490]: same here

File: modules/conversion-center/base/widget-contact-button-base.php (hunk 1 above)
Comment (willhowat) [reply modules/conversion-center/base/widget-contact-button-base.php:1435]: Hi @mserino this configuration is as far as I'm aware required across all variations.
What we're seeing here and may not be apparent just through the changes view, is a local variable used to control where this border is applied as it always has to be applied to the first control in this group, but visibility of some controls is optional. So we set this to show by default and then null it out as we render the controls.

If at some point this needs to move into configuration we can add a config item that sets the initial state, but at this time it does not vary across variants, unless there is one I've not seen

Comment (mserino) [reply modules/conversion-center/base/widget-contact-button-base.php:1435]: makes sense, thanks!

**Comment by willhowat** [comment]
Closing in favour of combined PR

**Pull request closed without merging**